MapboxNavigation/ImageDownloader.swift

```diff
@@ -45,9 +48,11 @@ class ImageDownloader: NSObject, ReentrantImageDownloader, URLSessionDataDelegat
             operation = activeOperation
         } else {
             operation = operationType.init(request: request, in: self.urlSession)
-            self.operations[url] = operation
+            accessQueue.async(flags: .barrier) {
+                self.operations[url] = operation
```

Comment:
Is it necessary to do a weakSelf/strongSelf dance here? (×2)

Reply:
No, I don't believe so. The implicit retain ensures self (and the access queue) exist long enough for the block to be executed. The block is transient, so when it completes self would be released.

Reply:
it's also worth noting that this object is owned by the ImageRepository, which is a `static let` singleton.